src/ot_tracer_provider.erl

```diff
@@ -46,6 +47,17 @@ register_tracer(Name, Vsn) ->
             false
     end.
 
+-spec register_application_tracer(atom()) -> boolean().
+register_application_tracer(Name) ->
+    try
+        {ok, Vsn} = application:get_key(Name, vsn),
+        {ok, Modules} = application:get_key(Name, modules),
+        gen_server:call(?MODULE, {register_tracer, Name, Vsn, Modules})
```

Comment:
This is to serialize it through the tracer provider. Otherwise with concurrent calls you could end up with some modules in an application pointing to one tracer and some pointing to another. This just makes it safe.